File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>
namespace log_line {
    std::string message(const std::string& m) {
        return m.substr(m.find(':') + 2); // substring after ": "
    }
    std::string log_level(const std::string& m) {
        return m.substr(1, m.find(']') - 1); // substring between "[" and "]"
    }
    std::string reformat(const std::string& m) {
        return message(m) + " (" + log_level(m) + ")";
    }
} // namespace log_line

File: solutions/csharp/log-levels/1/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        return logLine[(logLine.IndexOf(':') + 1)..].Trim();
    }

    public static string LogLevel(string logLine)
    {
        return logLine[1..logLine.IndexOf(']')].ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}
